lxd/api_cluster.go

```diff
@@ -535,6 +535,10 @@ func clusterPutJoin(d *Daemon, r *http.Request, req api.ClusterPut) response.Res
 				return response.SmartError(err)
 			}
 
+			localHTTPSAddress = req.ServerAddress
+		} else if util.IsWildCardAddress(localHTTPSAddress) {
```

Comment:
should this be an `else if` or just a separate if/

Reply:
It's slightly more efficient this way.

The block above calls [IsAddressCovered](https://github.com/canonical/lxd/blob/main/lxd/util/net.go#L164) which returns true in cases where either the `core` and `cluster` address are the same, or `core` is a wildcard. 

If the additional condition was a separate `if` block, then we would always run through the wildcard logic twice, whereas with an `else if` we only check the address again if the first condition fails.

Reply:
Thanks, we will backport this to 5.21